File: Design Patterns/Behavioral/Strategy/Case 2/Java/Challenge1/src/PhoneCameraApp.java
abstract class PhoneCameraApp {
    private ShareBehavior share;
    public void take(){

    }

    abstract public void edit();

    public void save(){

    }

    public void setShare(ShareBehavior share) {
        this.share = share;
    }

    public void preformShare(){
        share.share();
    }
}

File: Design Patterns/Behavioral/Strategy/Case 2/PHP/PhoneCameraApp.php
<?php
namespace PHP;
require_once("Behaviors/ShareBehavior.php");
use PHP\Behaviors\ShareBehavior;

abstract class PhoneCameraAPP{

    private ShareBehavior $shareBehavior;

    function getShareBehavior():ShareBehavior{
        return $this->shareBehavior;
    }

    function setShareBehavior(ShareBehavior $shareBehavior){
        $this->shareBehavior=$shareBehavior;
    }

    abstract function edit();

    function performShare(){
        $this->shareBehavior->share();
    }

    function take(){
        print("Picture taken");
    }

    function save(){
        print("Picture Saved");
    }
}